pymeasure/instruments/siglenttechnologies/siglent_spdbase.py

```diff
@@ -0,0 +1,291 @@
+
+        :type: str
+        """
+    )
+
+    fw_version = Instrument.measurement(
+        "SYST:VERS?",
+        """Read the software version of the instrument.
+
+        :type: str
+        """
+    )
+
+    system_status_code = Instrument.measurement(
+        "SYST:STAT?",
+        """Read the system status register.
+
+        :type: :class:`.SystemStatusCode`
+        """,
+        validator=truncated_range,
+        values=[0, 1024],
+        get_process=lambda v: SystemStatusCode(int(v, base=16)),
+    )
+
+    local_lockout = Instrument.setting(
+        "%s",
+        """``True`` disables local interface, ``False`` enables it.
+        """,
+        validator=strict_discrete_set,
+        values={True: "*LOCK", False: "*UNLOCK"},
+        map_values=True
+    )
+
+    save_settings = Instrument.measurement(
+        "*SAV %d",
+        """Save the current configuration in non-volatile memory.
+
+        :type: int
+        """,
+        validator=strict_discrete_set,
+        values=[1, 2, 3, 4, 5]
+    )
+
+    recall_settings = Instrument.measurement(
+        "*RCL %d",
+        """Recall the saved configuration from non-volatile memory.
+
+        :type: int
+        """,
+        validator=strict_discrete_set,
+        values=[1, 2, 3, 4, 5]
+    )
+
+    selected_channel = Instrument.control(
+        "INST?", "INST %s",
+        """Control the selected channel of the instrument.
+
+        :type: :class:`.SPDChannel`
+
+        """,
+        validator=strict_discrete_set,
+        values={1: "CH1"},
+        map_values=True,
+        dynamic=True
+    )
+
+    set_4W_mode = Instrument.setting(
+        "MODE:SET %s",
+        """Control the 4-Wire Sense Mode of the instrument.
+        """,
+        validator=strict_discrete_set,
+        values={False: "2W", True: "4W"},
+        map_values=True
+    )
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter,
+            usb=dict(write_termination='\n',
+                     read_termination='\n'),
+            tcpip=dict(write_termination='\n',
+                       read_termination='\n'),
+            **kwargs
+        )
+
+    def save_config(self, index):
+        """Save the current config to memory.
+
+        :param index:
+            string: index of the location to save the configuration
+        :returns: self
+        """
+        self.save_settings = index
+
+        return self
+
+    def recall_config(self, index):
+        """Recall a config from memory.
+
+        :param index:
+            string: index of the location from which to recall the configuration
+        :returns: self
+        """
+        self.recall_settings = index
+
+        return self
+
+
+class SPDChannel(object):
+    channel = 1
+
+    voltage = Instrument.measurement(
+        f"MEAS:VOLT? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
```

Comment:
Does this work as expected?

The property creators work at class level, therefore, all channels will be "CH1".

What you can do, is to define a (**not** f-string) `"MEAS:VOLT? CH{channel}"` and in `SPDChannel.write` you do `self.instrument.write(command.format(channel=self.channel))`. This way, you replace at instance level the channel name.

Reply:
Thanks for this one. I was slightly worried that was what I had done here. Good to have some confirmation. Changes are in the works